project/cmake/modules/FindOpenGLES.cmake

```diff
@@ -1,51 +1,48 @@
-# - Try to find OpenGLES2
-# Once done this will define
+#.rst:
+# FindOpenGLES
+# ------------
+# Finds the OpenGLES2 library
 #
-#  OPENGLES_FOUND        - system has OpenGLES2
-#  OPENGLES_INCLUDE_DIR  - the GLES2 include directory
-#  OPENGLES_LIBRARIES    - Link these to use OpenGLES2
+# This will will define the following variables::
+#
+# OPENGLES_FOUND - system has OpenGLES
+# OPENGLES_INCLUDE_DIRS - the OpenGLES include directory
+# OPENGLES_LIBRARIES - the OpenGLES libraries
+# OPENGLES_DEFINITIONS - the OpenGLES definitions
 
 find_package(EMBEDDED)
-find_package(PkgConfig)
 
 if(PKG_CONFIG_FOUND AND NOT PLATFORM STREQUAL "raspberry-pi")
-  pkg_check_modules(OPENGLES glesv2)
-  if(NOT OPENGLES_FOUND AND NOT EMBEDDED_FOUND STREQUAL "EMBEDDED_FOUND-NOTFOUND")
+  pkg_check_modules(PC_OPENGLES glesv2)
+  if(NOT OPENGLES_FOUND AND EMBEDDED_FOUND)
     set(CMAKE_PREFIX_PATH ${EMBEDDED_FOUND} ${CMAKE_PREFIX_PATH})
   endif()
 endif()
 
-
-if(NOT OPENGLES_FOUND)
-  if("${CORE_SYSTEM_NAME}" STREQUAL "ios")
-    find_library(OPENGLES_gl_LIBRARY NAMES OpenGLES PATHS ${CMAKE_OSX_SYSROOT}/System/Library PATH_SUFFIXES Frameworks NO_DEFAULT_PATH)
-    set(OPENGLES_INCLUDE_DIRS ${OPENGLES2_gl_LIBRARY}/Headers)
-    set(OPENGLES_egl_LIBRARY ${OPENGLES2_gl_LIBRARY})
-  else()
-    find_path(OPENGLES_INCLUDE_DIRS GLES2/gl2.h)
-    find_library(OPENGLES_gl_LIBRARY NAMES GLESv2)
-    find_library(OPENGLES_egl_LIBRARY NAMES EGL)
-  endif()
-
-  include(FindPackageHandleStandardArgs)
-  find_package_handle_standard_args(OpenGLES DEFAULT_MSG
-    OPENGLES_INCLUDE_DIRS OPENGLES_gl_LIBRARY OPENGLES_egl_LIBRARY)
-
-  set(OPENGLES_LIBRARIES ${OPENGLES2_gl_LIBRARY} ${OPENGLES2_egl_LIBRARY})
-  set(GLES_FOUND 1 CACHE INTERNAL "GLES")
-  set(OPENGLES_FOUND 1 CACHE INTERNAL "GLES")
-  mark_as_advanced(GLES_FOUND OPENGLES_FOUND)
+if(NOT CORE_SYSTEM_NAME STREQUAL ios)
+  find_path(OPENGLES_INCLUDE_DIR GLES2/gl2.h
+                                 PATHS ${PC_OPENGLES_INCLUDEDIR})
```

Comment:
imho it doesn't make sense to use find_path on a perfectly valid pkg-config result.